src/hooks/posters.ts

```diff
@@ -71,24 +72,55 @@ export const usePosters = (posterHallId: string) => {
   const fuseVenues = useMemo(
     () =>
       new Fuse(filteredPosterVenues, {
-        keys: ["poster.title", "poster.authorName", "poster.categories"],
+        keys: [
+          "name",
+          "poster.title",
+          "poster.authorName",
+          "poster.categories",
+        ],
+        threshold: 0.2, // 0.1 seems to be exact, default 0.6: brings too distant if anyhow related hits
+        ignoreLocation: true, // default False: True - to search ignoring location of the words.
+        findAllMatches: true,
       }),
     [filteredPosterVenues]
   );
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery)
-      return filteredPosterVenues.slice(0, displayedPostersCount);
+    const normalizedSearchQuery = searchQuery.trim();
+
+    if (!normalizedSearchQuery) return filteredPosterVenues;
+
+    const matchEntries = breakStringWithQuotesBySpaces(normalizedSearchQuery);
+
+    if (!matchEntries) return filteredPosterVenues;
 
     return fuseVenues
-      .search(searchQuery)
-      .slice(0, displayedPostersCount)
+      .search({
+        $and: matchEntries.map((x) => ({
+          $or: [
+            { name: x },
+            { "poster.title": x },
+            { "poster.authorName": x },
+            { "poster.categories": x },
+            // @debt Rewrite the search query to match the $and type
+          ] as Record<string, string>[],
+        })),
```

Comment:
What is the relevance of this `@debt` comment here? It's ambiguous currently, since there already seems to be an `$and` used here?

Reply:
I believe this related to what I called out in https://github.com/sparkletown/sparkle/pull/1460#discussion_r641295362

Which was fixed in 5668677